src/haz3lcore/dynamics/elaborator.re

```diff
@@ -51,222 +51,237 @@ let exp_binop_of: Term.UExp.op_bin => (Typ.t, (_, _) => DHExp.t) =
       ((e1, e2) => BinStringOp(string_op_of(op), e1, e2)),
     );
 
+/* Wrap: Handles cast insertion and non-empty-hole wrapping
+   for elaborated expressions */
+let wrap = (u, mode, self, d: DHExp.t): option(DHExp.t) =>
+  switch (Statics.error_status(mode, self)) {
+  | NotInHole(_) =>
+    switch (mode) {
+    | Syn => Some(d)
+    | SynFun =>
+      /* Things in function position get cast to an arrow type */
```

Comment:
nit: get cast to the matched arrow type. maybe make the `_` case an Arrow case and assert in the fall through case that it should never happen in this and the cases below